pandas/core/generic.py

```diff
@@ -7381,7 +7381,13 @@ def _clip_with_one_bound(self, threshold, method, axis, inplace):
         # GH 40420
         # Treat missing thresholds as no bounds, not clipping the values
         if is_list_like(threshold):
-            fill_value = np.inf if method.__name__ == "le" else -np.inf
+            method_name_le = method.__name__ == "le"
+            if is_datetime64_any_dtype(self.dtype):
+                fill_value = Timestamp.max if method_name_le else Timestamp.min
```

Comment:
for dt64tz shouldn't fill_value be tzaware?

Reply:
Is it possible to create a timezone aware Timestamp ts where ts > Timestamp.max would be true?

The fill value is not show in the result, it is just a placeholder

Reply:
> Is it possible to create a timezone aware Timestamp ts where ts > Timestamp.max would be true?

no, that comparison would always raise TypeError

Reply:
Hm ok, then we could use the max case for timzeones too